packages/@aws-cdk/aws-vpcv2-alpha/lib/ipam.ts

```diff
@@ -0,0 +1,428 @@
+ * @property {string} IP_V6 - Represents the IPv6 address family.
+ * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ec2-ipampool.html#cfn-ec2-ipampool-addressfamily
+ */
+export enum AddressFamily {
+  /**
+   * Represents the IPv4 address family.
+   * Allowed under public and private pool.
+   */
+  IP_V4 = 'ipv4',
+
+  /**
+   * Represents the IPv6 address family.
+   * Only allowed under public pool.
+   */
+  IP_V6 = 'ipv6',
+}
+
+/**
+ * Properties for creating an IPAM pool.
+ *
+ * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ec2-ipampool.html
+ *
+ * @interface IpamPoolProps
+ * @extends {PoolOptions}
+ * @property {string} ipamScopeId - The ID of the IPAM scope.
+ */
+export interface IpamPoolProps extends PoolOptions {
+  /**
+   * Scope id where pool needs to be created
+   */
+  readonly ipamScopeId: string;
+
+  /**
+   * IPAM resource name
+   * @default autogenerated by CDK
+   */
+  readonly ipamPoolName?: string;
+
+}
+
+/**
+ * Defining the type for allowed AWS service values under IPAM pool option
+ */
+type AwsServiceName = 'ec2'
+
+/**
+ * Options for configuring an IPAM pool.
+ *
+ * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ec2-ipampool.html
+ *
+ * @interface PoolOptions
+ * @property {AddressFamily} addressFamily - The address family of the pool (ipv4 or ipv6).
+ * @property {CfnIPAMPool.ProvisionedCidrProperty[]} [provisionedCidrs] - Information about the CIDRs provisioned to the pool.
+ * @property {string} [locale] - The locale (AWS Region) of the pool.
+ * @property {IpamPoolPublicIpSource} [publicIpSource] - The IP address source for pools in the public scope.
+ * @property {string} [awsService] - Limits which AWS service can use the pool.
+ */
+export interface PoolOptions{
+
+  /**
+   * addressFamily - The address family of the pool (ipv4 or ipv6).
+   */
+  readonly addressFamily: AddressFamily;
+
+  /**
+   * [provisionedCidrs] - Information about the CIDRs provisioned to the pool.
+   * @default - No CIDRs are provisioned
+   * @see CfnIPAMPool.ProvisionedCidrProperty
+   */
+  readonly provisionedCidrs?: CfnIPAMPool.ProvisionedCidrProperty[];
+
+  /**
+   * [locale] - The locale (AWS Region) of the pool.
+   * @default - The AWS Region of the stack
+   */
+  readonly locale?: string;
+
+  /**
+   * [publicIpSource] - The IP address source for pools in the public scope.
+   * Only used for IPv6 address
+   * byoip | amazon
+   * @default amazon
+   */
+  readonly publicIpSource?: IpamPoolPublicIpSource;
+
+  /**
+  * Limits which service in AWS that the pool can be used in.
+  *
+  * "ec2", for example, allows users to use space for Elastic IP addresses and VPCs.
+  *
+  * @see http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ec2-ipampool.html#cfn-ec2-ipampool-awsservice
+  *
+  * @default - No service
+  */
+  readonly awsService?: AwsServiceName;
```

Comment:
This value is required in case of assigning Ipv6 address in the IPAM pool

Reply:
so do you think the customer have to provide it, or it is something we can do on behalf of the customer

Reply:
checking back on this in console, this is something that customer could choose to configure in ipv4 as well, so it should be configurable at the customer end, since we have only `amazon` owned ipv6 address support this field is required, with byoip it can be set to null.